src/libraries/Native/Unix/System.Native/pal_log.m

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_log.h"
+#import <Foundation/Foundation.h>
+
+void SystemNative_Log (uint8_t* buffer, int32_t length)
+{
+    NSString *msg = [[NSString alloc] initWithBytes: buffer length: length encoding: NSUTF16LittleEndianStringEncoding];
```

Comment:
How does error handling work in Objective C? Does this throw exceptions? What is the usual way to deal with it?

Reply:
`initWithBytes` returns nil for invalid data, then all messages (`[obj message]`) are no-op or return 0/nil/null. The only thing - `strlen` won't be ok with NULL, let me fix it.